compiler/src/test/scala/dx/translator/cwl/ProcessTranslatorTest.scala

```diff
@@ -0,0 +1,151 @@
+      case (version, _) if Language.parse(version) == Language.CwlV1_2 => ()
+      case _ =>
+        throw new Exception(
+            s"""source code does not appear to be a CWL document of a supported version
+               |${cwlFile}""".stripMargin
+        )
+    }
+    val (process, schemas) = parser.parseFile(cwlFile) match {
+      case ParserResult(Some(process: Process), _, _, schemas) => (process, schemas)
+      case other =>
+        throw new Exception(s"expected CWL document to contain a CommandLineTool, not ${other}")
+    }
+    val cwlBundle: CwlBundle = CwlBundle.create(process)
+    val callableTranslator = ProcessTranslator(
+        cwlBundle,
+        cwlSchemas = schemas,
+        locked = false,
+        defaultRuntimeAttrs = Map.empty,
+        reorgAttrs = DefaultReorgSettings(false),
+        perWorkflowAttrs = Map.empty,
+        defaultScatterChunkSize = 1,
+        useManifests = false,
+        instanceTypeSelection = InstanceTypeSelection.Static
+    )
+
+    val depOrder: Vector[Process] = cwlBundle.sortByDependencies
+
+    // translate processes
+    val (allCallables, sortedCallables) =
+      depOrder.foldLeft((Map.empty[String, Callable], Vector.empty[Callable])) {
+        case ((allCallables, sortedCallables), callable) =>
+          val isPrimary = callable.name == cwlBundle.primaryProcess.name
+          val translatedCallables =
+            callableTranslator.translateProcess(callable, allCallables, isPrimary = isPrimary)
+          (
+              allCallables ++ translatedCallables.map(c => c.name -> c).toMap,
+              sortedCallables ++ translatedCallables
+          )
+      }
+
+    sortedCallables.map {
+      case Application(name, _, _, _, _, _, document, _, _, _, _, _) =>
+        name -> document.getDocContents
+      case Workflow(name, _, _, _, document, _, _, _, _, _, _) =>
+        name -> document.getDocContents
+    }.toMap
+  }
+
+  "ProcessTranslator" should "render same cwl code for every unchanged process and different if there are changes" in {
+    val standAloneProcessDoc1 = getProcessDocContent("count-lines1-wf")
+    val standAloneProcessDoc2 = getProcessDocContent("count-lines1-wf-dup")
+    standAloneProcessDoc1("parseInt-tool.cwl") should equal(
+        standAloneProcessDoc2("parseInt-tool.cwl")
+    )
+
+    // task wc-tool.cwl changed
+    standAloneProcessDoc1("wc-tool.cwl") should not equal standAloneProcessDoc2("wc-tool.cwl")
+    standAloneProcessDoc1("count-lines1-wf_common") should not equal standAloneProcessDoc2(
+        "count-lines1-wf_common"
+    )
+    standAloneProcessDoc1("count-lines1-wf_outputs") should not equal standAloneProcessDoc2(
+        "count-lines1-wf_outputs"
+    )
+    standAloneProcessDoc1("count-lines1-wf") should not equal standAloneProcessDoc2(
+        "count-lines1-wf"
+    )
+
+  }
+
+  "ProcessTranslator" should "render different cwl code for changed common/outputs" in {
```

Comment:
Just as a convention: class name to be tested is written only for the first test in the series of tests. All subsequent tests are named just `it`.

Reply:
Thanks! I didn't notice that when copypasting. Changed as you suggested.